### ISSUE TITLE
[PROPOSAL] Add padding option to pooling layers

### ISSUE THREAD

**Issue by nedtaylor** [nedtaylor/athena#51]
### Reasoning

Same as what is seen in the `convXd_layer_type` layer types in athena currently (development branch commit 1b3ebea6382995070edc3d52b79a2ad22bc035f2 and onwards), pooling layer types should include the option to add padding.

### Prior Art

See `conv1d_layer_type` and other convolutional layers in athena.

See pytorch and tensorflow for examples where padding is included in pooling layers.
https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html
